lib/logstash/outputs/elasticsearch.rb

```diff
@@ -188,7 +188,7 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # Set the address of a forward HTTP proxy.
   # This used to accept hashes as arguments but now only accepts
   # arguments of the URI type to prevent leaking credentials.
-  config :proxy, :validate => :uri
+  config :proxy, :validate => :uri # but empty string is allowed
```

Comment:
Let's add a note to the `docs/index.asciidoc` entry on the proxy setting in that it's ok to have it empty and what the consequence is (same as no proxy). Even a note on how this is useful when you're using envionment variables with an example would be awesome!

Reply:
yy, circling back here. left this one unfinished - still appreciate the review.